manifests/service.pp

```diff
@@ -7,11 +7,37 @@
 #
 # * Justin Lambert <mailto:jlambert@letsevenup.com>
 #
-class kibana::service {
+class kibana::service (
+  $service_provider    = $::kibana::service_provider,
+) {
 
-  service { 'kibana':
-    ensure => running,
-    enable => true,
+  if $service_provider == 'init' {
```

Comment:
I think it's cleaner to install the init files in the install.pp as that takes care of the duplication of the service because of the dependency.